File: Javascript/Power-Of-Four.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function (n) {
  if (n == 0) {
    return false;
  }

  while (n % 4 == 0) {
    n = n >> 2;
  }

  return n == 1;
};

File: Rust/Power-Of-Four.rs
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n == 0 {
            return false;
        }

        let mut n = n;

        while n % 4 == 0 {
            n = n >> 2;
        }

        n == 1
    }
}
